crates/sp-executor/src/lib.rs

```diff
@@ -57,6 +57,8 @@ impl sp_runtime::BoundToRuntimeAppPublic for ExecutorKey {
 /// Header of transaction bundle.
 #[derive(Decode, Encode, TypeInfo, PartialEq, Eq, Clone, RuntimeDebug)]
 pub struct BundleHeader {
+    /// The hash of primary block at which the bundle was created.
+    pub primary_hash: PHash,
     /// The slot number.
     pub slot_number: u64,
```

Comment:
It doesn't seem that slot number is actually read anywhere (IDE wasn't able to find at least). Will we use it for something? BTW, it must be possible to extract it using `primary_hash` either way.

Reply:
I think we'll need the slot number when dealing with the bundle equivocation?

https://github.com/subspace/subspace/blob/8f119714d8642126595fd0404dc29c2c4bae7abe/cumulus/client/cirrus-executor/src/lib.rs#L651

Not sure how you can extract the slot number using `primary_hash`, there are possibly multiple slots that are mapped to the same primary hash.

Reply:
`primary_hash` means very specific block. And that block contains time slot in the header, so I don't see how there can be multiple time slots mapped to the same block.

Reply:
You are mixing the slot usage, now we depend on the slot for two things: 1. publish a puzzle and farmers try to solve this puzzle to produce a block, 2. executors try to produce a bundle on each new slot. Not every slot will have a new block produced, but there are multiple bundles that can be produced against the same primary block in the meantime, in which case we have multiple bundles produced on different slots but on top of the same primary block.

Reply:
Now I get it, thanks for explaining